### PR TITLE
Copy and move API for content and media

### PR DESCRIPTION
**Pull request by kjac**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description

This PR adds copy and move APIs for Content and move API for media (media cannot be copied) to the management API.

### Testing this PR

Use Swagger or Postman (@kjac has a collection) to copy and move content and to move media using the new endpoints. 

Any changes made should be immediately visible in the current backoffice, provided that the items will actually load (some property editors no longer work with the new serialization).

It is advised to disable automatic launch of the site when testing, specially when testing the media, because image cropper likely won't play nice because of config serialization. This can be achieved by setting `"launchBrowser": false` in `launchSettings.json`.

Look at the unit tests in this PR for inspiration for edge case operations to test - for example it shouldn't be possible to move an item beneath itself.

While the PR does not handle granular permissions (see FIXME comments), the general setup of allowed child types does work and should be tested as well.

### CODE REVIEW

**Review by bergmania — COMMENTED** [review]
We decided to call all things Id instead of Key in the openAPI contract. 

I can just fix it before testing.

File: src/Umbraco.Core/Services/MediaEditingService.cs [hunk 1]
```diff
@@ -101,4 +108,6 @@ private ContentEditingOperationStatus Save(IMedia media, int userId)
             return ContentEditingOperationStatus.Unknown;
         }
     }
+
+    private async Task<int> GetUserIdAsync(Guid userKey) => await _userIdKeyResolver.GetAsync(userKey) ?? Constants.Security.SuperUserId;
```
Comment (bergmania) [inline src/Umbraco.Core/Services/MediaEditingService.cs:112]: Everywhere else this seems to be such that the userKey is inputed to the service layer. Maybe we should align.

No matter what, I dont like the idea about automatically using SuperUser

Comment (bergmania) [reply src/Umbraco.Core/Services/MediaEditingService.cs:112]: We agreed on huddle that we on a separate PR will fix this, so _userIdKeyResolver.GetAsync throws instead of being nullable

**Review by bergmania — APPROVED** [review]
